### ISSUE TITLE
Hard-to-interpret error if a directory/file named "latex" exists in the source directory

### ISSUE THREAD

**Issue by manfredlotz** [tectonic-typesetting/tectonic#754]
This is Fedora 33 and tectonic 0.4.1

I run `tectonic x.tex` and get

```
note: this is a BETA release; ask questions and report bugs at https://tectonic.newton.cx/
Running TeX ...
warning: 4-byte read failed
caused by: Is a directory (os error 21)
error: something bad happened inside TeX; its output follows:

===============================================================================
===============================================================================
error: the TeX engine had an unrecoverable error
caused by: could not undump 1 4-byte item(s) from latex
```

When I move the file into a different directory and run `tectonic x.tex` no error at all, i.e. the pdf gets built.

I have no idea what's going on.

**Comment by pkgw** [comment]
So if I'm understanding correctly, the build succeeds in one directory, but fails in another? Maybe the bad directory is read-only or something?

Would you be comfortable obtaining an attaching an `strace` log to the failing run? That might help in this case.

**Comment by manfredlotz** [comment]
It was definitely not a r/o file system. Now, it is working again. No idea what happened. Next time I will run `strace` rightaway.

Nevertheless, do you have any idea what those error messages tell me?

**Comment by dbuscher** [comment]
I have the exact same problem, but on OS-X/brew (did not realise that moving the files to a new directory would work but it does!). Is there some equivalent of strace I can run?

**Comment by dbuscher** [comment]
OK, I have worked out the immediate cause: if I remove the directory called "latex" from the working directory, it works, creating a new directory called "latex" causes it to fail. Latex file as follows:

\documentclass{article}
\begin{document}
This is a test
\end{document}

**Comment by manfredlotz** [comment]
Yep, **good catch**. Perhaps I had a directory `latex` where things did not work some days ago. 

At least, when creating a directory `latex` I get the error which caused me to open the issue.

**Comment by ralismark** [comment]
Looks like this issue has been fixed -- this could probably be closed now.

**Comment by manfredlotz** [comment]
When I clone the repository and build tectonic in master I see the same error as almost 2 months ago. What could I do to verify that it has been fixed?

**Comment by pkgw** [comment]
@manfredlotz The issue identified here should be fixed in master, yes. To confirm that you're running the latest code, I guess you could try creating a minimal dummy document and verifying that it builds (1) as a minumum working example and (2) when you create a `latex` directory to trigger the bug that I believe should be fixed?

If both of those builds work, then you must be encountering an issue that is similar but not quite the same.

**Comment by pkgw** [comment]
This should be fixed as of the 0.5.0 release — please file a new issue if this problem (or something superficially similar) appears.

**Comment by manfredlotz** [comment]
Did the following test. `cargo install tectonic` which installed 0.5.2

Then doing `tectonic x.tex` which worked ok. Then `mkdir latex` and `tectonic x.tex` which also looked ok. 

So, I consider this as fixed. Thanks a lot!
